spotirec.py

```diff
@@ -998,53 +1002,68 @@ def parse():
             rec.rec_params[x.split('=')[0]] = x.split('=')[1]
 
 
-args = parser.parse_args()
+def init():
+    global logger, conf, api, sp_oauth, rec, headers
+
+    # Logging handler
+    logger = log.Log()
+    if args.verbose:
+        logger.set_level(log.VERBOSE)
+    elif args.quiet:
+        logger.set_level(log.WARNING)
+    elif args.debug:
+        logger.set_level(log.DEBUG)
+
+    if args.suppress_warnings:
+        logger.suppress_warnings(True)
+
+    logger.verbose('initialising')
+    logger.debug(f'log level: {logger.LEVEL} ({log.LOG_LEVELS[logger.LEVEL]})')
+    logger.debug(f'suppress warnings: {logger.SUPPRESS_WARNINGS}')
+
+    # Config handler
+    conf = sp_conf.Config()
+    conf.set_logger(logger)
+
+    # API handler
+    api = sp_api.API()
+    api.set_logger(logger)
+    api.set_conf(conf)
 
-# Logging handler
-logger = log.Log()
-if args.verbose:
-    logger.set_level(log.VERBOSE)
-elif args.quiet:
-    logger.set_level(log.WARNING)
-elif args.debug:
-    logger.set_level(log.DEBUG)
-
-if args.suppress_warnings:
-    logger.suppress_warnings(True)
-
-logger.verbose('initialising')
-logger.debug(f'log level: {logger.LEVEL} ({log.LOG_LEVELS[logger.LEVEL]})')
-logger.debug(f'suppress warnings: {logger.SUPPRESS_WARNINGS}')
-
-# Config handler
-conf = sp_conf.Config()
-conf.set_logger(logger)
-
-# API handler
-api = sp_api.API()
-api.set_logger(logger)
-api.set_conf(conf)
-
-# OAuth handler
-sp_oauth = oauth2.SpotifyOAuth()
-sp_oauth.set_logger(logger)
-sp_oauth.set_conf(conf)
-sp_oauth.set_api(api)
-
-headers = {'Content-Type': 'application/json',
-           'Authorization': f'Bearer {get_token()}'}
-
-# Recommendation object
-if args.load_preset:
-    rec = load_preset(args.load_preset[0])
-else:
-    rec = recommendation.Recommendation()
-    parse()
-rec.set_logger(logger)
-
-logger.debug(f'args: {args}')
-
-if __name__ == '__main__':
-    recommend()
-    if args.log:
-        logger.log_file()
+    # OAuth handler
+    sp_oauth = oauth2.SpotifyOAuth()
+    sp_oauth.set_logger(logger)
+    sp_oauth.set_conf(conf)
+    sp_oauth.set_api(api)
+
+    headers = {'Content-Type': 'application/json',
+               'Authorization': f'Bearer {get_token()}'}
+
+    # Recommendation object
+    if args.load_preset:
+        rec = load_preset(args.load_preset[0])
+        rec.set_logger(logger)
+    else:
+        rec = recommendation.Recommendation()
+        rec.set_logger(logger)
+        parse()
+
+    logger.debug(f'args: {args}')
+
+
+logger = None
+conf = None
+api = None
+sp_oauth = None
+rec = None
+headers = None
+parser = create_parser()
+args = parser.parse_args()
+# Allows import for tests
+if not any('unittest' in arg for arg in sys.argv):
+    if __name__ == '__main__':
+        setup_config_dir()
+        init()
+        recommend()
+        if args.log:
+            logger.log_file()
```

Comment:
What's the reason for using `sys.argv` instead of argparse? It seems weird to use both.